src/cpp/session/modules/rmarkdown/NotebookCache.hpp

```diff
@@ -0,0 +1,75 @@
+/*
+ * NotebookCache.hpp
+ *
+ * Copyright (C) 2009-16 by RStudio, Inc.
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
+
+// A notebook .Rmd is accompanied by a sidecar .Rnb.cached folder, which has
+// the following structure:
+//
+// - foo.Rmd
+// + .foo.Rnd.cached
```

Comment:
That's a good question. I hadn't thought it was necessary because we already control the name of the root folder (so we can invalidate existing caches just by changing the way we form the root folder name, or by including a version identifier in its name), but now that I think it through it might be helpful to be able to e.g. expire older cache structures without having to keep old root directory formatting rules around. I'll add this.